httpie/sessions.py

```diff
@@ -79,6 +79,13 @@ def update_headers(self, request_headers: RequestHeadersDict):
             if name == 'User-Agent' and value.startswith('HTTPie/'):
                 continue
 
+            if name == 'Cookie':
+                for cookie in value.split(';'):
+                    name, value = cookie.split("=")
+                    self['cookies'][name.strip()] = {'value': value.strip()}
+                del request_headers['Cookie']
+                continue
+
```

Comment:
Isn’t there an existing implementation of the `Cookie` header value parsing? I’d imagine `requests` or the standard library having one.

Reply:
The parser should also gracefully handle situations when the header contains invalid value (e.g., `Cookie: hello` or `Cookie: ;;;===`)—people often intentionally send invalid requests with HTTPie to test the server’s behavior.

Reply:
If you end up using a custom implementation, then I’d probably move it to a function (and add some unittests for it).